pycbc/io/live.py

```diff
@@ -159,38 +162,57 @@ def __init__(self, ifos, coinc_results):
 
-            valid_end = int(len(qtilde) - data_readers[ifo].trim_padding)
-            valid_start = int(valid_end - data_readers[ifo].blocksize * data_readers[ifo].sample_rate)
-            seg = slice(valid_start, valid_end)
-            snr = snr[seg]
-            snr *= norm
-            delta_t = 1.0 / data_readers[ifo].sample_rate
-            start = data_readers[ifo].start_time
-            snr = TimeSeries(snr, delta_t=delta_t, epoch=start)
-            snr.save(filename, group='%s/snr' % ifo)
-            stilde.psd.save(filename, group='%s/psd' % ifo)
-        GraceDb().writeFile(graceid, filename)
+        # compute SNR time series
+        self.upload_snr_series = kwargs['upload_snr_series']
+        if self.upload_snr_series:
+            data_readers = kwargs['data_readers']
+            bank = kwargs['bank']
+            htilde = bank[self.template_id]
+            self.snr_series = {}
+            self.snr_series_psd = {}
+            for ifo in self.ifos:
+                stilde = data_readers[ifo].overwhitened_data(htilde.delta_f)
+                norm = 4.0 * htilde.delta_f / (htilde.sigmasq(stilde.psd) ** 0.5)
+                qtilde = zeros((len(htilde)-1)*2, dtype=htilde.dtype)
+                correlate(htilde, stilde, qtilde)
+                snr = qtilde * 0
+                ifft(qtilde, snr)
+
+                valid_end = int(len(qtilde) - data_readers[ifo].trim_padding)
+                valid_start = int(valid_end - data_readers[ifo].blocksize * data_readers[ifo].sample_rate)
+                seg = slice(valid_start, valid_end)
+                snr = snr[seg]
+                snr *= norm
+                delta_t = 1.0 / data_readers[ifo].sample_rate
+                start = data_readers[ifo].start_time
+                snr = TimeSeries(snr, delta_t=delta_t, epoch=start)
+                self.snr_series[ifo] = snr
+                self.snr_series_psd[ifo] = stilde.psd
+
+                # store the on-source slice of the series into the XML doc
+                snr_onsource_time = coinc_results['foreground/%s/end_time' % ifo] - snr.start_time
+                snr_onsource_dur = lal.REARTH_SI / lal.C_SI
+                onsource_idx = int(snr_onsource_time * snr.sample_rate)
+                onsource_start = onsource_idx - int(snr.sample_rate * snr_onsource_dur / 2)
+                onsource_end = onsource_idx + int(snr.sample_rate * snr_onsource_dur / 2)
+                onsource_slice = slice(onsource_start, onsource_end + 1)
+                snr_lal = snr[onsource_slice].lal()
+                snr_lal.name = 'snr'
+                snr_lal.sampleUnits = ''
+                snr_xml = _build_series(
+                        snr_lal, (u"Time", u"Time,Real,Imaginary"), None,
+                        'deltaT', 's')
+                snr_node = outdoc.childNodes[-1].appendChild(snr_xml)
+                eid_param = ligolw_param.from_pyvalue(
+                        u"event_id", str(sngl_event_id_map[ifo]))
```

Comment:
Remove the `str()`: this should be an ilwdchar. This code produces:

    <Param Type="lstring" Name="event_id:param">sngl_inspiral:event_id:1</Param>

whereas BAYESTAR will look for:

    <Param Type="ilwd:char" Name="event_id:param">sngl_inspiral:event_id:1</Param>